[CI] TransformContinuousIT testContinuousEvents failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/bgbdl2jxi6xyi/tests/:x-pack:plugin:transform:qa:multi-node-tests:javaRestTest/org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT/testContinuousEvents

**Reproduction line:**
```
gradlew ':x-pack:plugin:transform:qa:multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinuousEvents" -Dtests.seed=F3BBB10D61182B19 -Dtests.locale=sl -Dtests.timezone=Jamaica -Druntime.java=19
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT&tests.test=testContinuousEvents

**Failure excerpt:**
```
java.lang.AssertionError: transform [continuous-terms-on-date-pivot-test] does not progress, iteration: 9, state: indexing, reason: null
Expected: is a value greater than <2023-02-19T21:39:54.531821500Z>
     but: <2023-02-19T21:39:52.307Z> was less than <2023-02-19T21:39:54.531821500Z>

  at __randomizedtesting.SeedInfo.seed([F3BBB10D61182B19:B17C7CB25EF060D]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.lambda$waitUntilTransformsProcessedNewData$3(TransformContinuousIT.java:489)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1154)
  at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.waitUntilTransformsProcessedNewData(TransformContinuousIT.java:484)
  at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinuousEvents(TransformContinuousIT.java:263)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1589)

```

This might just be a slow windows VM, I see only windows but only one linux error: 

https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P28D&search.timeZoneId=Europe/Berlin&tests.container=org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT&tests.test=testContinuousEvents#

If I look at the rocky linux one, I see this CI instance had multiple issues and long runtimes after branching `8.6` and wasn't stable for some time when the failure happened.

That makes it 4 windows errors, further inspecting the log:

```
java.lang.AssertionError: transform [continuous-terms-on-date-pivot-test] does not progress, iteration: 9, state: indexing, reason: null
10:24:47     Expected: is a value greater than <2023-02-22T09:24:06.951924400Z>
10:24:47          but: <2023-02-22T09:24:00.948Z> was less than <2023-02-22T09:24:06.951924400Z>
```

The message includes`iteration 9`, the other have `5`, `6` and another `9`. There are 10 iterations in total(counting from `0`). That supports the assumption of a slow running process. A problem in the test would more likely happen in the low iterations.

Further we can reconstruct timings from the log messages by inspecting the log lines containing:

` [testContinuousEvents] wait until transform reaches timestamp_millis: 2023-02-22T09:19:13.484193Z`

If I look at the times from https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+main+multijob+platform-support-windows/GRADLE_TASK=checkPart2,os=windows-2012-r2/678/consoleFull

| time | duration in s |
|--------|---------------|
|19:12| |
|19:25| 13 |
|19:40| 15 |
|20:08| 28 |
|20:35 | 27 |
|21:21 | 46 |
|21:51 | 30|
|22:39 | 48 |
|23:14 | 35 |
|24:05 | 51 |
| `timeout` | |

It seems that we were lucky that it didn't timeout earlier. The tests uses `assertBusy` with a `30s` timeout. However as multiple data scenarios are run in parallel, it didn't fail exactly after `30s`.

## Action

I will revert the test mute, but also double the `assertBusy` to `60s`. In addition I will improve the error message to indicate a potential performance problem, likely _not_ originating from transform.

## Retrospective

I think this mute was unjustified, quoting the SOP: 

> failed multiple times recently.

The test failed 4 times with days in between. In addition tests shouldn't be muted if

> It's clearly some kind of transitive issue. ... This includes things like request timeouts, ...

In addition all failures happened on windows, same platform, although different flavors. Interesting: all failures are either `9:07 AM` or `9:07 PM`. Is there some windows maintenance task running at that time?

Anyway, maybe the SOP should be improved. Still the judgement was too quick in this case. The _entire_ sweet got disabled, although just 1 part of it had a _timeout_. We've lost important coverage.

--- a/x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java
+++ b/x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java
@@ -9,13 +9,13 @@ package org.elasticsearch.xpack.transform.integration.continuous;
 
 import org.apache.http.entity.ContentType;
 import org.apache.http.entity.StringEntity;
-import org.apache.lucene.tests.util.LuceneTestCase;
 import org.elasticsearch.client.Request;
 import org.elasticsearch.client.RequestOptions;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.util.concurrent.ThreadContext;
 import org.elasticsearch.common.xcontent.support.XContentMapValues;
+import org.elasticsearch.core.CheckedRunnable;
 import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.core.Tuple;
 import org.elasticsearch.xcontent.XContentBuilder;
@@ -85,9 +85,9 @@ import static org.hamcrest.Matchers.notNullValue;
  *          to check that optimizations worked
  *      - repeat
  */
-@LuceneTestCase.AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/93900")
 public class TransformContinuousIT extends TransformRestTestCase {
 
+    public static final int MAX_WAIT_TIME_ONE_ITERATION_SECONDS = 60;
     private List<ContinuousTestCase> transformTestCases = new ArrayList<>();
 
     @Before
@@ -267,21 +267,8 @@ public class TransformContinuousIT extends TransformRestTestCase {
 
             // test the output
             for (ContinuousTestCase testCase : transformTestCases) {
-                try {
-                    testCase.testIteration(run, modifiedEvents);
-                } catch (AssertionError testFailure) {
-                    throw new AssertionError(
-                        "Error in test case ["
-                            + testCase.getName()
-                            + "]."
-                            + "If you want to mute the test, please mute ["
-                            + testCase.getClass().getName()
-                            + "] only, but _not_ ["
-                            + this.getClass().getName()
-                            + "] as a whole.",
-                        testFailure
-                    );
-                }
+                int thisRun = run;
+                assertForTestCase(() -> testCase.testIteration(thisRun, modifiedEvents), testCase);
             }
         }
     }
@@ -482,29 +469,33 @@ public class TransformContinuousIT extends TransformRestTestCase {
             delay,
             iteration
         );
-        for (ContinuousTestCase testCase : transformTestCases) {
-            assertBusy(() -> {
-                var stats = getTransformStats(testCase.getName());
-                Object lastSearchTimeObj = XContentMapValues.extractValue("checkpointing.last_search_time", stats);
-                assertThat(lastSearchTimeObj, is(notNullValue()));
-                long lastSearchTime = (long) lastSearchTimeObj;
-                assertThat(
-                    "transform ["
-                        + testCase.getName()
-                        + "] does not progress, iteration: "
-                        + iteration
-                        + ", state: "
-                        + stats.get("state")
-                        + ", reason: "
-                        + stats.get("reason"),
-                    Instant.ofEpochMilli(lastSearchTime),
-                    is(greaterThan(waitUntil))
-                );
-                // assert a checkpoint isn't in progress
-                Object state = XContentMapValues.extractValue("state", stats);
-                assertThat(state, is(equalTo("started")));
-            }, 30, TimeUnit.SECONDS);
-        }
+        assertBusy(() -> {
+            for (ContinuousTestCase testCase : transformTestCases) {
+                assertForTestCase(() -> {
+                    var stats = getTransformStats(testCase.getName());
+                    Object lastSearchTimeObj = XContentMapValues.extractValue("checkpointing.last_search_time", stats);
+                    assertThat(lastSearchTimeObj, is(notNullValue()));
+                    long lastSearchTime = (long) lastSearchTimeObj;
+                    assertThat(
+                        Strings.format(
+                            "Timeout [%ds] waiting for transform [%s] to finish next checkpoint, "
+                                + "iteration [%d], state [%s], reason in case of failure [%s], last search time [%d]",
+                            MAX_WAIT_TIME_ONE_ITERATION_SECONDS,
+                            testCase.getName(),
+                            iteration,
+                            stats.get("state"),
+                            stats.get("reason"),
+                            lastSearchTime
+                        ),
+                        Instant.ofEpochMilli(lastSearchTime),
+                        is(greaterThan(waitUntil))
+                    );
+                    // assert a checkpoint isn't in progress
+                    Object state = XContentMapValues.extractValue("state", stats);
+                    assertThat(state, is(equalTo("started")));
+                }, testCase);
+            }
+        }, MAX_WAIT_TIME_ONE_ITERATION_SECONDS, TimeUnit.SECONDS);
     }
 
     private void addTestCaseIfNotDisabled(ContinuousTestCase testCaseInstance) {
@@ -532,6 +523,24 @@ public class TransformContinuousIT extends TransformRestTestCase {
         logger.info("deletePipeline: {}", pipelineId);
         Request putPipeline = new Request("DELETE", "/_ingest/pipeline/" + pipelineId);
         assertAcknowledged(client().performRequest(putPipeline));
+    }
+
+    private void assertForTestCase(CheckedRunnable<Exception> codeBlock, ContinuousTestCase testCase) throws Exception {
+        try {
+            codeBlock.run();
+        } catch (AssertionError testFailure) {
+            throw new AssertionError(
+                "Error in test case ["
+                    + testCase.getName()
+                    + "]."
+                    + "If you want to mute the test, please mute ["
+                    + testCase.getClass().getName()
+                    + "] only, but _not_ ["
+                    + this.getClass().getName()
+                    + "] as a whole.",
+                testFailure
+            );
+        }
     }
 
     @Override

